File: Programmers/2/138476. 귤 고르기/귤 고르기.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <unordered_map>

using namespace std;

int solution(int k, vector<int> tangerine) {
    int answer = 0;
    
    unordered_map<int, int> count;
    for (int t : tangerine)
        count[t]++;

    vector<int> freq;
    for (auto &p : count)
        freq.push_back(p.second);

    sort(freq.begin(), freq.end(), greater<int>());

    int total = 0;
    for (int f : freq)
    {
        total += f;
        answer++;
        
        if (total >= k) break;
    }    
    
    return answer;
}

File: Programmers/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;

class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        int total = 0;
        
        Map<Integer, Integer> countMap = new HashMap<>();
        
        for (int t : tangerine) {
            countMap.put(t, countMap.getOrDefault(t, 0) + 1);
        }

        List<Integer> counts = new ArrayList<>(countMap.values());
        
        counts.sort(Collections.reverseOrder());
        
        for (int c : counts) {
            total += c;
            answer++;
            
            if (total >= k) break;
        }
        
        return answer;
    }
}